subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBase.java

```diff
@@ -687,6 +690,30 @@ public List<SubscriptionBillingEvent> getSubscriptionBillingEvents(final Version
         }
     }
 
+
+    private DateTime alignToNextBCD(final Plan curPlan, final PlanPhase curPlanPhase, final DateTime originalDate, final SubscriptionCatalog catalog, final Integer bcdLocal, final InternalTenantContext context) throws SubscriptionBaseApiException, CatalogApiException {
+
+        final BillingAlignment billingAlignment = catalog.billingAlignment(new PlanPhaseSpecifier(curPlan.getName(), curPlanPhase.getPhaseType()),
+                                                                           originalDate, originalDate);
+
+        final int accountBillCycleDayLocal = apiService.getBCD(context);
+        Integer bcd = bcdLocal;
+        if (bcd == null) {
```

Comment:
Cannot be `null`

Reply:
What about [this](https://github.com/killbill/killbill/pull/1663/files#diff-6c9485b2174859a0d155b69e387e4b868e364c2608ba08fd0a2db9c49101fc46R645) 🤔 ?

Reply:
Weird, I remember reading a different piece of code 🙃  In this case, it's always `null`?

Reply:
I think the caller needs to maintain the current per-subscription BCD as it goes through billing event, which is why I have this parameter, but right not implemented - let's keep this open.

Reply:
I implemented it in https://github.com/killbill/killbill/pull/1663/commits/4479e18ee70a81cf8aca0310ec48884d0daf3768